tests/integration/src/test/resources/http-response-timeout-config.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns:mule-ss="http://www.mulesoft.org/schema/mule/spring-security"
+        xmlns:http="http://www.mulesoft.org/schema/mule/http" xmlns="http://www.mulesoft.org/schema/mule/core"
+        xmlns:spring="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:ss="http://www.springframework.org/schema/security"
+        xsi:schemaLocation="http://www.mulesoft.org/schema/mule/spring-security http://www.mulesoft.org/schema/mule/spring-security/current/mule-spring-security.xsd
+http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
+http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+http://www.mulesoft.org/schema/mule/spring-security http://www.mulesoft.org/schema/mule/spring-security/current/mule-spring-security.xsd
+http://www.springframework.org/schema/security http://www.springframework.org/schema/security/spring-security-3.0.xsd">
+
+    <http:request-config name="HTTP_Request" host="127.0.0.1" port="${port}" responseTimeout="${delay}">
+        <http:basic-authentication username="username" password="password"/>
+    </http:request-config>
+
+    <http:request-config name="preemptive" host="127.0.0.1" port="${port}" responseTimeout="${delay}">
+        <http:basic-authentication username="username" password="password" preemptive="true"/>
```

Comment:
Since the only change is this boolean, I think it would make more sense to use a system property here and parameterised the test with it.